onprc_ehr/src/org/labkey/onprc_ehr/ONPRC_EHRController.java

```diff
@@ -512,4 +519,151 @@ public void setStartingId(Integer startingId)
             _startingId = startingId;
         }
     }
+
+    public static class SnomedForm
+    {
+        private String _subset;
+        private String _snomed;
+
+        public String getSubset()
+        {
+            return _subset;
+        }
+
+        public void setSubset(String subset)
+        {
+            _subset = subset;
+        }
+
+        public String getSnomed()
+        {
+            return _snomed;
+        }
+
+        public void setSnomed(String snomed)
+        {
+            _snomed = snomed;
+        }
+    }
+
+    @RequiresPermission(ReadPermission.class)
+    public class GetSnomedAction extends ReadOnlyApiAction<SnomedForm>
+    {
+        @Override
+        public ApiResponse execute(SnomedForm form, BindException errors)
+        {
+            SQLFragment sql = new SQLFragment();
+            if (form.getSubset() != null)
+            {
+                sql.append("SELECT sn.code FROM ehr_lookups.snomed sn ");
+                sql.append("LEFT JOIN ehr_lookups.snomed_subset_codes ssc ");
+                sql.append("ON sn.code = ssc.code ");
+                sql.append("WHERE ssc.primaryCategory = ? AND sn.container = ? AND soundex(sn.meaning) = soundex(?)");
```

Comment:
soundex fuzzy matching is a pretty neat usage here

Reply:
yeah pretty cool. We could break it down even further to word-by-word if we want. postgres has a similar function